crates/proof-of-sql/src/proof_primitive/dory/pack_scalars.rs

```diff
@@ -0,0 +1,926 @@
+}
+
+/// Returns the size of the largest committable column.
+///
+/// # Arguments
+///
+/// * `committable_columns` - A reference to the committable columns.
+fn get_max_committable_column_length(committable_columns: &[CommittableColumn]) -> usize {
+    committable_columns
+        .iter()
+        .map(|column| column.len())
+        .max()
+        .unwrap_or(0)
+}
+
+/// Returns the minimum value of a column as an Fr.
+///
+/// # Arguments
+///
+/// * `column` - A reference to the committable column.
+pub fn get_min_as_fr(column: &CommittableColumn) -> F {
+    match column {
+        CommittableColumn::SmallInt(_) => i16::min_as_fr(),
+        CommittableColumn::Int(_) => i32::min_as_fr(),
+        CommittableColumn::BigInt(_) => i64::min_as_fr(),
+        CommittableColumn::Int128(_) => i128::min_as_fr(),
+        CommittableColumn::Decimal75(_, _, _) => <[u64; 4]>::min_as_fr(),
+        CommittableColumn::Scalar(_) => <[u64; 4]>::min_as_fr(),
+        CommittableColumn::VarChar(_) => <[u64; 4]>::min_as_fr(),
+        CommittableColumn::Boolean(_) => <[u64; 4]>::min_as_fr(),
+        CommittableColumn::TimestampTZ(_, _, _) => i64::min_as_fr(),
+    }
+}
+
+/// Returns a repeated bit table vector that duplicated the
+/// bit table for each element by the num_sub_commits.
+///
+/// # Arguments
+///
+/// * `bit_table` - A reference to the bit table.
+/// * `num_sub_commits` - The number of sub commits.
+fn get_repeated_bit_table(bit_table: &[u32], num_sub_commits: usize) -> Vec<u32> {
+    bit_table
+        .iter()
+        .flat_map(|&value| std::iter::repeat(value).take(num_sub_commits))
+        .collect()
+}
+
+/// Returns the number of commits needed for the packed_msm function.
+///
+/// # Arguments
+///
+/// * `committable_columns` - A reference to the committable columns.
+/// * `offset` - The offset to the data.
+/// * `num_columns` - The number of generators used for msm.
+pub fn get_num_of_sub_commits_per_full_commit(
+    committable_columns: &[CommittableColumn],
+    offset: usize,
+    num_columns: usize,
+) -> usize {
+    // Committable columns may be different sizes, get the max size and add offset.
+    let max_column_length = get_max_committable_column_length(committable_columns) + offset;
+
+    // Number of scalar vectors that the size of the generators n.
+    // Each scalar will be used to call the packed_msm function.
+    (max_column_length + num_columns - 1) / num_columns
+}
+
+/// Modifies the signed sub-commits by adding the offset to the sub-commits.
+///
+/// # Arguments
+///
+/// * `commits` - A reference to the signed sub-commits.
+/// * `committable_columns` - A reference to the committable columns.
+/// * `num_full_commits` - The number of full commits.
+/// * `num_sub_commits_per_full_commit` - The number of sub commits needed for each full commit for the packed_msm function.
+#[tracing::instrument(name = "pack_scalars::modify_commits (gpu)", level = "debug", skip_all)]
+pub fn modify_commits(
```

Comment:
Yeah, I struggled with this one. The motivation to put this in `pack_scalars` was because the function needs to know how exactly how the scalar offsets get packed. Any updates to the way the scalars are packed will also change this code. On the other hand, the function isn't involved in packing any scalars.

Do you think it would make more sense in the `dory_commitment_helper_gpu` module?